scc/src/main.rs

```diff
@@ -49,18 +60,46 @@ fn main() -> Result<(), Error> {
     Ok(())
 }
 
-fn setup_logger(cache_dir: &Path) -> Result<(), Error> {
+pub fn exists(file : &Path) -> bool {
+    fs::metadata(file).is_ok()
+}
+
+fn setup_logger(cache_dir: &Path, date_time_logger : bool) -> Result<(), Error> 
+{
+    let parent_dir  = cache_dir.parent().unwrap();
+    let mut log_name : String = "redscript".to_owned();
+    
+    if date_time_logger == true
+    {
+        log_name.push_str("_");
+        let time_create = OffsetDateTime::now_local().unwrap().format(&Rfc3339Format).unwrap();
+        log_name.push_str(&time_create);
+
+        log_name = log_name.replace("-", ".").replace(":", "-")[..29].to_string();
+        log_name.replace_range(20..21, "_");
+    }
+    
+    log_name.push_str(".log");
+
+    // Log directory make
+    let log_dir = parent_dir.join("logs");
+
+    if !exists(&log_dir)
```

Comment:
`exists` is already defined for paths, so you should be able to just write `if log_dir.exists()`